### PR TITLE
drt netlist refactor for memory optimization

### PR DESCRIPTION
**Pull request by habibayassin**

### CODE REVIEW

File: src/drt/src/db/obj/frInst.h [hunk 1]
```diff
@@ -45,12 +46,13 @@ class frInst : public frRef
...
-  frInst(const frString& name, frMaster* master)
-      : name_(name), master_(master), pinAccessIdx_(0), toBeDeleted_(false)
+  frInst(frMaster* master, odb::dbInst* inst)
+      : master_(master), inst_(inst), pinAccessIdx_(0), toBeDeleted_(false)
   {
   }
   // getters
-  const frString& getName() const { return name_; }
+  odb::dbInst* getDbInst() const { return inst_; }
+  const frString getName() const { return inst_->getName(); }
```
Comment (rovinski) [inline src/drt/src/db/obj/frInst.h:55]: Passed by value so does not need to be `const`

File: src/drt/src/db/obj/frNet.h [hunk 2]
```diff
@@ -50,9 +51,10 @@ class frNet : public frBlockObject
 {
  public:
   // constructors
-  frNet(const frString& in)
+  frNet()
```
Comment (rovinski) [inline src/drt/src/db/obj/frNet.h:54]: If you can construct an `frNet` with no name and no `dbNet*` then how can you disambiguate two empty `frNet`s?

If this is to support fake nets then I would suggest removing the default constructor and instead having a `frNet(bool, bool)` constructor to specify `isFakeVSSNet_` and `isFakeVDDNet_`.

File: src/drt/src/db/obj/frNet.h [hunk 3]
```diff
@@ -231,8 +267,11 @@ class frNet : public frBlockObject
       max = std::max(max, CLOCK_NETS_ABS_PRIORITY);
     absPriorityLvl = max;
   }
-  bool isSpecial() const { return isSpecial_; }
-  void setIsSpecial(bool s) { isSpecial_ = s; }
+  bool isSpecial() const
+  {
+    return db_net_->isSpecial();
+    ;
```
Comment (rovinski) [inline src/drt/src/db/obj/frNet.h:273]: remove extra `;`

File: src/drt/src/io/io.cpp [hunk 4]
```diff
@@ -176,7 +156,7 @@ void io::Parser::setObstructions(odb::dbBlock* block)
       continue;
     }
     frLayerNum layerNum = tech_->name2layer[layerName]->getLayerNum();
-    auto blkIn = make_unique<frBlockage>();
+    auto blkIn = make_unique<frBlockage>(blockage->getBBox());
```
Comment (rovinski) [inline src/drt/src/io/io.cpp:159]: Does it make more sense to store the `dbObstruction` than the bbox of the obstruction?

File: src/drt/src/io/io.cpp [hunk 5]
```diff
@@ -821,7 +801,7 @@ void io::Parser::setNets(odb::dbBlock* block)
         }
       }
     }
-    netIn->setType(net->getSigType());
+    // netIn->setType(net->getSigType());
```
Comment (rovinski) [inline src/drt/src/io/io.cpp:804]: Remove commented code

File: src/drt/test/fixture.cpp [hunk 6]
```diff
@@ -83,8 +86,10 @@ void Fixture::addLayer(frTechObject* tech,
 
 void Fixture::setupTech(frTechObject* tech)
 {
-  auto db = odb::dbDatabase::create();
+  // auto db = odb::dbDatabase::create();
```
Comment (rovinski) [inline src/drt/test/fixture.cpp:89]: Remove

File: src/drt/test/fixture.cpp [hunk 7]
```diff
@@ -83,8 +86,10 @@ void Fixture::addLayer(frTechObject* tech,
...
   db_tech = odb::dbTech::create(db);
+  db->getTech()->setDbUnitsPerMicron(1000);
+  db->getTech()->setManufacturingGrid(10);
```
Comment (rovinski) [inline src/drt/test/fixture.cpp:92]: Is this duplicate?

File: src/drt/src/io/io.cpp [hunk 8]
```diff
@@ -1810,9 +1771,8 @@ void io::Parser::setMacros(odb::dbDatabase* db)
             }
           }
         }
-        auto blkIn = make_unique<frBlockage>();
+        auto blkIn = make_unique<frBlockage>(obs);
```
Comment (habibayassin) [inline src/drt/src/io/io.cpp:1774]: > Does it make more sense to store the `dbObstruction` than the bbox of the obstruction?
I am trying to be consistent with the object used in this code block which is of type bbox

File: src/drt/src/db/obj/frMaster.h [hunk 9]
```diff
@@ -78,13 +76,49 @@ class frMaster : public frBlockObject
...
+    frBoundary bound;
+    vector<Point> points;
+    points.push_back(Point(originX, originY));
+    points.push_back(Point(sizeX, originY));
+    points.push_back(Point(sizeX, sizeY));
+    points.push_back(Point(originX, sizeY));
+    bound.setPoints(points);
+    bounds.push_back(bound);
+    return bounds;
+  }
```
[the remark this answers is not in the document]
Comment (habibayassin) [reply src/drt/src/db/obj/frMaster.h:116]: I think it is a tradeoff between mem consumption and runtime cost...

**Pull request closed without merging**